### ISSUE TITLE
docs: Drawer needs clarification on a11y landmarks

### ISSUE THREAD

**Issue by kenn** [saadeghi/daisyui#4022]
### On which page do you see this issue?

https://admit-ai.com/

### Describe the issue

Drawer: https://daisyui.com/components/drawer/
Per accessibility requirements: "All page content should be contained by landmarks"
https://dequeuniversity.com/rules/axe/4.10/region

Running axe flags the hidden `<input type="checkbox" class="drawer-toggle">` used by the Drawer as content outside of landmarks (“All page content must be contained by landmarks”). The Drawer pattern requires the checkbox to be a direct child of .drawer, so it can’t be wrapped in a landmark without breaking the CSS sibling selectors

You can test it here:

```sh
npx @axe-core/cli https://admit-ai.com
```

Here's the basic structure from the original doc.

```jsx
<div className="drawer drawer-end">
  <input id="my-drawer-4" type="checkbox" className="drawer-toggle" />
  <div className="drawer-content">
    {/* Page content here */}
    <label htmlFor="my-drawer-4" className="drawer-button btn btn-primary">Open drawer</label>
  </div>
  <div className="drawer-side">
    <label htmlFor="my-drawer-4" aria-label="close sidebar" className="drawer-overlay"></label>
    <ul className="menu bg-base-200 text-base-content min-h-full w-80 p-4">
      {/* Sidebar content here */}
      <li><a>Sidebar Item 1</a></li>
      <li><a>Sidebar Item 2</a></li>
    </ul>
  </div>
</div>
```

### What browsers are you seeing the problem on?

All browsers

**Comment by saadeghi** [comment]
The checkbox is not the cause of warning. It's using a div as direct child of `<body>`.

Add a `role="main"` to the div:
```jsx
<div className="drawer drawer-end" role="main">
```
Or use `<main>` tag instead
```jsx
<main className="drawer drawer-end">
```
And it should be the direct child of `<body>`.  

The examples in docs is using div because we can't actually decide or predict how you're using this drawer. It depends on the framework.  
It can be a direct child of `<body>` (in that case you may get such warning) or it can be nested in a wrapper injected by framework.  
Other than that, you may already have a `<main>` tag in your app (from a default layout file from framework) so we can't provide examples with `<main>` tag and people will end up with a nested `<main>` inside another `<main>` which is wrong.

daisyUI doesn't require using divs. Only class name is important so you can use `<main>`, `<aside>`, `<section>` etc based on your need.

Let me know if you have a question.
